src/libcore/raw.rs

```diff
@@ -93,7 +93,32 @@
 #[repr(C)]
 #[derive(Copy, Clone)]
 #[allow(missing_debug_implementations)]
+#[unstable(feature = "raw", issue = "27751")]
 pub struct TraitObject {
     pub data: *mut (),
     pub vtable: *mut (),
 }
+
+/// Returns the value of the discriminant for the enum variant in `v`.
```

Comment:
It is probably worth adding a warning that discriminants may not compare in the correct order if the enum is `#[repr(u64)]` due to overflow.